bumble/device.py

```diff
@@ -5069,13 +5093,15 @@ def on_link_key(self, bd_addr, link_key, key_type):
             )
 
             utils.cancel_on_event(
-                self, 'flush', self.update_keys(str(bd_addr), pairing_keys)
+                self, Device.EVENT_FLUSH, self.update_keys(str(bd_addr), pairing_keys)
             )
 
         if connection := self.find_connection_by_bd_addr(
             bd_addr, transport=PhysicalTransport.BR_EDR
         ):
+            connection.link_key = link_key
```

Comment:
Saving link_key is a little bit weird here. In spec, link key is almost always not associated with connection but an address.Saving link_key is a little bit weird here. In spec, link key is almost always not associated with a connection but an address.

If we worry about optional keystore might not be set, we may make it required and put an `WriteOnlyMemoryKeystore` default implementation which only records keys without providing them.

Reply:
You are 100% right. I had added `link_key` here to go along with the existing `link_key_type`, but now that you mention it, neither of those really belong here. They only belong in the key store. 
I removed `link_key`, but I'm leaving `link_key_type` as-is for now, because it seems that Avatar relies on it, but we should fix the Avatar test to look up the link key in the key store rather than in a property of the connection. Once that's fixed, we can remove the property.